connector/countconnector/connector.go

```diff
@@ -0,0 +1,125 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package countconnector // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/countconnector"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const scopeName = "otelcol/countconnector"
+
+// count can count spans, data points, or log records and emit
+// the count onto a metrics pipeline.
+type count struct {
+	Config
+	consumer.Metrics
+	component.StartFunc
+	component.ShutdownFunc
+}
+
+func (c *count) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (c *count) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	countMetrics := pmetric.NewMetrics()
+	for i := 0; i < td.ResourceSpans().Len(); i++ {
+		resourceSpan := td.ResourceSpans().At(i)
+		countResource := countMetrics.ResourceMetrics().AppendEmpty()
+		resourceSpan.Resource().Attributes().CopyTo(countResource.Resource().Attributes())
+
+		countScope := countResource.ScopeMetrics().AppendEmpty()
+		countScope.Scope().SetName(scopeName)
+
+		var count int
```

Comment:
```suggestion
		count := 0
```

To be consistent with ConsumeMetrics and ConsumeLogs